### ISSUE TITLE
Cannot open files with square brackets in the name

### ISSUE THREAD

**Issue by mathstuf** [scrooloose/nerdtree#391]
$subject.

**Comment by briandonahue** [comment]
Seeing the same thing.  Could this be new/recent behavior?  I swear this used to work in an older version

**Comment by mathstuf** [comment]
Could be a Vim bug as well.

**Comment by briandonahue** [comment]
Not sure how I would test that, but vim opens the files fine via e, tabe, and drag/drop...

**Comment by briandonahue** [comment]
Possibly possibly related to #420 ?

**Comment by mathstuf** [comment]
Possibly. ISTR it being escaped brackets as well. Ill have to check my install logs to see if Vim got bumped around this time. Did you update Vim lately?

**Comment by briandonahue** [comment]
I don't know what *ISTR* is... but yes, this may be a newer version of Vim than I had before, though I'm not sure offhand. It *is* a new machine, and I switched to vundle and updated my plugins.  It says it's 7.4, "MS-Windows 32-bit GUI version with OLE Support"

**Comment by mathstuf** [comment]
> ISTR
"I seem to remember"

> 7.4
There should be a patch level with this (e.g., 7.4.475).

**Comment by briandonahue** [comment]
```
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov  3 2014 09:04:09)
MS-Windows 32-bit GUI version with OLE support
Included patches: 1-493
Compiled by digitectNO@SPAMdancingpaper.com:http://SteveHallArchitecture.com
```

**Comment by justfalter** [comment]
I'm seeing this on 8.0, as well.
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Sep 14 2016 03:42:03)
MacOS X (unix) version
Included patches: 1-3
Compiled by travis@Traviss-Mac-659.local
```

**Comment by justfalter** [comment]
From my testing, the problem seems to occur when a file or directory name contains a [, any number of characters, and ].

The following filenames are problematic for me:
```
test[foo].txt
test[[foo].txt
test[].txt
```

The following filenames open just fine:
```
test[foo.txt
testfoo].txt
test]foo[.txt
```

**Comment by przepompownia** [comment]
Confirmed on `[nerdtree].txt` (NVIM 0.1.6-dev, Vim 8.0.22).

**Comment by Nausx** [comment]
I'm using Slackware 14.2 at the moment with the default package of gvim and i'm having the same issue. This concern directory opening as well. In case you wonder if it's a vim bug, it doesn't happen with the default file explorer. and it only affect files and directories with matching brackets. i.e. : `this[file]name.whatever` wont open but `this[filename.whatever` will.

**Comment by mathstuf** [comment]
I think the filename is probably treated as a regex somewhere and not escaped properly. Other regex-special characters probably also cause issues in nerdtree.

**Comment by lifecrisis** [comment]
Directly related to #680. This is a confirmed bug.

**Comment by MrMino** [comment]
+1, I suspect that this issue originetates from the fact that those names aren't escaped at some point, and they are interpreted as bash wildcards.

**Comment by lifecrisis** [comment]
@MrMino, the problem was identified in #680.

**Comment by PhilRunninger** [comment]
@mathstuf @briandonahue @justfalter @przepompownia @Nausx @MrMino , would you be willing to test the pull request #868 I've put together for this issue? Details are in the PR. Thanks.

**Comment by briandonahue** [comment]
I am on holiday until 8/27, but will be happy to check it out after that if it still needs testing.

**Comment by PhilRunninger** [comment]
Fixed in #868.
